### PR TITLE
CSS Scroll Improvements

### PR DESCRIPTION
**Pull request by DennisTsar**

### CODE REVIEW

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt [hunk 1]
```diff
@@ -24,18 +24,37 @@ fun StyleScope.scrollBehavior(scrollBehavior: ScrollBehavior) {
 
 // region Scroll snap
 // See https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type
-class ScrollSnapType private constructor(private val value: String) : StylePropertyValue {
+sealed class ScrollSnapType private constructor(private val value: String) : StylePropertyValue {
     override fun toString() = value
 
+    private class Keyword(value: String) : ScrollSnapType(value)
+    class Axis internal constructor(value: String) : ScrollSnapType(value)
+    private class WithStrictness(axis: Axis, strictness: Strictness) : ScrollSnapType("$axis $strictness")
```
Comment (bitspittle) [inline frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt:32]: Totally optional: `AxisWithStrictness`

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt [hunk 2]
```diff
@@ -57,15 +85,25 @@ class ScrollSnapMode private constructor(private val value: String) : StylePrope
 
     companion object {
         // Keyword
+        @Deprecated(
+            "Use `ScrollSnapType.Strictness.Mandatory` instead.",
```
Comment (bitspittle) [inline frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt:89]: Stale comments I think. `of(axis, Mandatory)` instead

**Review by bitspittle — APPROVED** [review]
Only nits

File: frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt (hunk 2 above)
Comment (DennisTsar) [reply frontend/compose-html-ext/src/jsMain/kotlin/com/varabyte/kobweb/compose/css/Scroll.kt:89]: The idea behind the deprecations is that a previous use of `Modifier.scrollSnapType(ScrollSnapAxis.Both, ScrollSnapMode.Mandatory)` ends up with a deprecation `ReplaceWith` for each parameter, which after being applied will become the valid `Modifier.scrollSnapType(ScrollSnapType.Both, ScrollSnapType.Strictness.Mandatory)`

**Pull request merged**